scripts/script-tag.js

```diff
@@ -139,9 +148,16 @@
                     if (continueSelling && continueSelling == 'true') {
 
                         // finding a button with type submit as the button will be on the same level as the input field so using siblings
-                        const preorderButton = variantTagInput.siblings("#hc_preorderButton");
+                        const preorderButton = variantTagInput.siblings("#hc_preorderButton, .hc_preorderButton");
                         const cartForm = variantTagInput.parent();
-                        const date = productType === 'Pre-Order' ? preOrderDate : productType === 'Back-Order' && backOrderDate;
+                        let date = productType === 'Pre-Order' ? preOrderDate : productType === 'Back-Order' && backOrderDate;
+
+                        // if the date is of past then making it empty
+                        let now = new Date();
+                        now.setHours(0, 0, 0, 0)
+                        if (new Date(date) < now) {
```

Comment:
What if date is empty?

Reply:
Sir, if the date is empty and the condition becomes false and we have handled the empty date case before adding it to the cart.